lib/WebService/Async/Segment.pm

```diff
@@ -0,0 +1,239 @@
+package WebService::Async::Segment;
+
+use strict;
+use warnings;
+
+use Net::Async::HTTP;
+use IO::Async::Loop;
+use Scalar::Util qw(blessed);
+use URI::Template;
+use JSON::MaybeUTF8 qw(encode_json_utf8 decode_json_utf8);
+use Syntax::Keyword::Try;
+use Log::Any qw($log);
+use Date::Utility;
+
+use WebService::Async::Segment::Customer;
+
+use constant SEGMENT_BASE_URL => 'https://api.segment.io/v1/';
+
+our $VERSION = '0.001';
```

Comment:
Why not make it as a subclass of `IO::Async::Notifier` ?

Reply:
Make a subclass.